src/Serialization/Compact.php

```diff
@@ -41,13 +41,19 @@ public function __construct(
 
     /**
      * @param string $headersJson
+     *
      * @return HeaderParameter\ParameterInterface[]
+     * @throws \InvalidArgumentException
      */
     protected function parseHeaders($headersJson)
     {
         $parameters = [];
         $headers = json_decode($headersJson, true);
 
+        if (!is_array($headers) || empty($headers)) {
```

Comment:
I think leave the `json_last_error_msg()` function out for now. It can easily be added later if more accurate messages are required, but removing it later if it is conflicting with other people's libraries is a bit of a problem. Future releases will require later versions of PHP which will make this available.